### PR TITLE
remove Kotlin's "Context Receiver" feature

### LINKED ISSUE #16863: [DEPRECATION]: Handle Kotlin context receivers
Need to handle/ or remove the experimental API:
> Phased removal of context receivers feature﻿
In Kotlin 1.6.20, we introduced context receivers as an Experimental feature. After listening to community feedback, we've decided not to continue with it. Instead, we plan to introduce a replacement in future Kotlin releases: context parameters. You can find the proposal for context parameters in the KEEP.

### (Optional) Anything else you want to share?

https://kotlinlang.org/docs/whatsnew-eap.html#phased-removal-of-context-receivers-feature

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
The feature will be removed and replaced with Context Parameters. It is unlikely that there will be any overlap period, so we need to remove our Context Receiver code, with the intention of replacing it at a later date

> * **2.0.20**: A warning is introduced for declarations that use
> context receivers.
> The warning should mention that the design of context receivers
> is about to change, according to the new design (add a link to the KEEP/YT).
> The safest workaround is to migrate your code to extensions.
>  * **2.1.20**: The implementation of context receivers is removed from the compiler.
> Effectively, this warning is turned into an error.
> Note that this part is most likely inevitable as the implementation
> of context parameters requires  quite extensive refactoring in the compiler,
> and there are significant chances that we won't have either
> context receivers or context parameters at that time.
>  [There is a slight chance that we'll have an early prototype of context parameters,
> but we are not counting on that.]
> * **2.2.0**: Context parameters are introduced under a new compiler flag.

https://youtrack.jetbrains.com/issue/KT-67119/Migration-warning-from-context-receivers-to-context-parameters

> **Phased removal of context receivers feature**
> In Kotlin 1.6.20, we introduced context receivers as an
> Experimental feature. After listening to community
> feedback, we've decided not to continue with it. Instead,
> we plan to introduce a replacement in future Kotlin
> releases: context parameters. You can find the proposal
> for context parameters in the KEEP.

https://kotlinlang.org/docs/whatsnew-eap.html#phased-removal-of-context-receivers-feature

**Deprecated:**
https://redirect.github.com/Kotlin/KEEP/blob/context-parameters/proposals/context-receivers.md
* https://redirect.github.com/Kotlin/KEEP/issues/259

**To be replaced by:**
https://redirect.github.com/Kotlin/KEEP/blob/context-parameters/proposals/context-parameters.md
* https://redirect.github.com/Kotlin/KEEP/issues/367

## Fixes

* Fixes #16863
* Reverts: 672b075a546e806e93f8e2c158749231c065a021
* Reverts: 1c01e4cb6f06a464f31fcc80a1907d58d1952c63

## Testing

* I briefly opened the app on the emulator and ran a full test suite.
  * No issues were found

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by criticalAY — APPROVED** [review]
Looks good!
I honestly did not expect Kotlin to remove this, it was actually useful

File: AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.kt [hunk 1]
```diff
@@ -1061,7 +1060,7 @@ open class Reviewer :
     override suspend fun updateCurrentCard() {
         val state = withCol {
             sched.currentQueueState()?.apply {
-                topCard.renderOutput(true)
+                topCard.renderOutput(this@withCol, reload = true, browser = false)
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.kt:1063]: Setting the value for browesr does not seems useful given that this was also the default value for the method in Card.kt

**Review by Arthur-Milchior — APPROVED** [review]
Feel free to click merge, or correct first.
Thanks for this boring work

File: AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/Reviewer.kt:1063]: There was an IDE warning to name the parameters, but sure

**Pull request merged**